app/utils/computed-helpers.js

```diff
@@ -47,24 +47,31 @@ export const computedSegmentedLink = function(property) {
 export const computedDateTimeSplit = function(property, segmentFormat, endProperty) {
   return computed(property, {
     get() {
-      if (this.constructor.toString() === 'model:event') {
-        return moment(this.get(property)).tz(this.timezone).format(getFormat(segmentFormat));
+      let timezone = '';
+      if (this.constructor.modelName === 'event') {
+        timezone = this.timezone;
       } else {
-        return moment(this.get(property)).format(getFormat(segmentFormat));
+        timezone = this.event.get('timezone');
```

Comment:
This is worse than before

Reply:
Idea is to write less code, not more

Reply:
So, make this work in least code possible, please

Reply:
I don't want to assign a variable outside if as it may crash the page of the other model inside an if condition.
For example:
```
let timezone = this.timezone;
if(this.constuctor.modelName !== 'event') {
	timezone = this.event.get('timezone');
}
```
This may break in session-speakers page, as it will first search for this.timezone.

Reply:
I don't want to assign a variable outside if as it may crash the page of the other model inside an if condition.
For example:
```
let timezone = this.timezone;
if(this.constuctor.modelName !== 'event') {
	timezone = this.event.get('timezone');
}
```
This may break in session-speakers page, as it will first search for this.timezone.